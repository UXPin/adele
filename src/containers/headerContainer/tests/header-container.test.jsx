import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import HeaderContainer from '../header-container';

it('renders HeaderContainer correctly', () => {
  const wrapper = mount(<HeaderContainer scroll={true} />);
  expect(wrapper).toMatchSnapshot();
});
