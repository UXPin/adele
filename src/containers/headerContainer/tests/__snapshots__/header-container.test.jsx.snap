// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HeaderContainer correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 6000;
  margin: 0 auto;
  width: 90%;
  background-image: url(test-file-stub);
  background-repeat: no-repeat;
  background-position: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: fixed;
  z-index: 99999;
  top: 0;
  margin-top: 0;
  width: 90%;
  height: 80px;
  background-color: white;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 100px;
}

.c2::before {
  content: 'by';
  display: block;
  position: absolute;
  top: 30px;
  left: 74px;
  opacity: 1;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 14px;
  color: #777777;
  font-weight: 300;
}

.c2::after {
  content: 'Adele';
  display: block;
  position: absolute;
  top: 25px;
  left: 10px;
  opacity: 1;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 21px;
  color: #222222;
  font-weight: bold;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c2 figcaption {
  opacity: 0;
  margin-left: 10px;
  margin-top: -3px;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 14px;
  color: #777777;
  font-weight: 300;
}

.c2 figcaption a {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid transparent;
  -webkit-transition: all 0.5s ease-in;
  transition: all 0.5s ease-in;
}

.c2 figcaption a:active,
.c2 figcaption a:visited,
.c2 figcaption a:link,
.c2 figcaption a:focus {
  color: #777777;
}

.c2 figcaption a:hover {
  border-bottom: 1px solid #777777;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 6px;
}

.c3 [title~='facebook'] span {
  padding: 0 0 1px 2px;
}

.c3 [title~='twitter'] span {
  padding: 0 0 0 2px;
}

.c3 [title~='linkedin'] span {
  padding: 0 0 0 2px;
}

.c3 [title~='github'] span {
  padding: 0 0 1px 2px;
}

.c3 [title~='github'] span svg {
  width: 20px;
  margin-left: -7px;
}

.c4 {
  display: inline-block;
  margin-left: 6px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  background: none;
  border: 1px solid #006cff;
  border-radius: 100%;
  -webkit-transition: background-color 0.3s ease-in;
  transition: background-color 0.3s ease-in;
  cursor: pointer;
  outline: none;
}

.c5 span {
  position: relative;
  display: block;
  width: 14px;
  height: 14px;
}

.c5 svg {
  display: block;
  -webkit-transition: fill 0.3s ease-in;
  transition: fill 0.3s ease-in;
  width: 10px;
  height: 14px;
}

.c5:hover,
.c5:focus {
  background-color: #006cff;
}

.c5:hover svg,
.c5:focus svg {
  fill: white;
}

.c6:focus {
  outline: 0;
}

.c6 > svg {
  width: 14px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c6 > svg:hover {
  fill: #006cff;
}

.c7:focus {
  outline: 0;
}

.c7 > svg {
  width: 13px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c7 > svg:hover {
  fill: #006cff;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c10:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c10:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c10:focus {
  outline: none;
  background-color: #3389ff;
}

.c11 {
  position: relative;
  outline: none;
  border: none;
  background: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  font-weight: 300;
  cursor: pointer;
}

.c11:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.c11:focus {
  color: #006cff;
}

.c8 {
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  margin: 150px auto 90px auto;
}

.c8 h1 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 230px;
  text-align: center;
  -webkit-letter-spacing: -13px;
  -moz-letter-spacing: -13px;
  -ms-letter-spacing: -13px;
  letter-spacing: -13px;
  text-shadow: 10px 10px 20px rgba(0,0,55,0.07);
}

.c8 span {
  display: block;
  position: relative;
  width: 500px;
  margin: -10px auto 0 auto;
  font-size: 26px;
  font-weight: 300;
  line-height: 1.3;
  color: #777777;
  text-align: center;
  text-shadow: none;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  margin-top: 40px;
}

.c9 > button:first-child {
  margin-right: 40px;
}

<HeaderContainer
  scroll={true}
>
  <styled.header
    id="header"
  >
    <header
      className="c0"
      id="header"
    >
      <TopBar
        scroll={true}
      >
        <styled.div
          id="top-bar"
          scroll={true}
        >
          <div
            className="c1"
            id="top-bar"
          >
            <styled.figure
              scroll={true}
            >
              <figure
                className="c2"
              >
                <Logo
                  tab={1}
                >
                  <a
                    href="https://www.uxpin.com?utm_source=adele.uxpin.com&utm_medium=upartner&utm_campaign=Adele"
                    rel="noopener noreferrer"
                    tabIndex={1}
                    target="_blank"
                    title="UXPin logo"
                  >
                    <SVGInline
                      classSuffix="-svg"
                      cleanup={Array []}
                      cleanupExceptions={Array []}
                      component="span"
                      svg="test-file-stub"
                    >
                      <span
                        className="SVGInline"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "test-file-stub",
                          }
                        }
                      />
                    </SVGInline>
                  </a>
                </Logo>
                <figcaption>
                  <a
                    href="http://uxpin.com"
                    rel="noopener noreferrer"
                    tabIndex={1}
                    target="_blank"
                    title="UXPin - The Design Systems Platform"
                  >
                    The Design Systems Platform
                  </a>
                </figcaption>
              </figure>
            </styled.figure>
            <Social
              networks={
                Array [
                  "twitter",
                  "facebook",
                  "linkedin",
                ]
              }
              tab={1}
              url="https://www.uxpin.com/adele"
            >
              <styled.ul>
                <ul
                  className="c3"
                >
                  <styled.li
                    title="github"
                  >
                    <li
                      className="c4"
                      title="github"
                    >
                      <styled.span
                        tabIndex={1}
                      >
                        <span
                          className="c5"
                          tabIndex={1}
                        >
                          <a
                            href="https://github.com/uxpin/adele"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <Icon
                              active={true}
                              color="#006cff"
                              i="test-file-stub"
                              in="blue"
                              rotate={0}
                              size="m"
                            >
                              <Styled(SVGInline)
                                color="#006cff"
                                data-active={true}
                                in="blue"
                                rotate={0}
                                size="m"
                                svg="test-file-stub"
                              >
                                <SVGInline
                                  className="c6"
                                  classSuffix="-svg"
                                  cleanup={Array []}
                                  cleanupExceptions={Array []}
                                  color="#006cff"
                                  component="span"
                                  data-active={true}
                                  in="blue"
                                  rotate={0}
                                  size="m"
                                  svg="test-file-stub"
                                >
                                  <span
                                    className="SVGInline c6"
                                    color="#006cff"
                                    dangerouslySetInnerHTML={
                                      Object {
                                        "__html": "test-file-stub",
                                      }
                                    }
                                    data-active={true}
                                    in="blue"
                                    rotate={0}
                                    size="m"
                                  />
                                </SVGInline>
                              </Styled(SVGInline)>
                            </Icon>
                          </a>
                        </span>
                      </styled.span>
                    </li>
                  </styled.li>
                  <styled.li
                    key="twitter"
                    title="twitter social share"
                  >
                    <li
                      className="c4"
                      title="twitter social share"
                    >
                      <styled.span
                        onClick={[Function]}
                        tabIndex={1}
                      >
                        <span
                          className="c5"
                          onClick={[Function]}
                          tabIndex={1}
                        >
                          <Icon
                            active={true}
                            color="#006cff"
                            i="test-file-stub"
                            in="blue"
                            rotate={0}
                            size="s"
                          >
                            <Styled(SVGInline)
                              color="#006cff"
                              data-active={true}
                              in="blue"
                              rotate={0}
                              size="s"
                              svg="test-file-stub"
                            >
                              <SVGInline
                                className="c7"
                                classSuffix="-svg"
                                cleanup={Array []}
                                cleanupExceptions={Array []}
                                color="#006cff"
                                component="span"
                                data-active={true}
                                in="blue"
                                rotate={0}
                                size="s"
                                svg="test-file-stub"
                              >
                                <span
                                  className="SVGInline c7"
                                  color="#006cff"
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "test-file-stub",
                                    }
                                  }
                                  data-active={true}
                                  in="blue"
                                  rotate={0}
                                  size="s"
                                />
                              </SVGInline>
                            </Styled(SVGInline)>
                          </Icon>
                        </span>
                      </styled.span>
                    </li>
                  </styled.li>
                  <styled.li
                    key="facebook"
                    title="facebook social share"
                  >
                    <li
                      className="c4"
                      title="facebook social share"
                    >
                      <styled.span
                        onClick={[Function]}
                        tabIndex={1}
                      >
                        <span
                          className="c5"
                          onClick={[Function]}
                          tabIndex={1}
                        >
                          <Icon
                            active={true}
                            color="#006cff"
                            i="test-file-stub"
                            in="blue"
                            rotate={0}
                            size="s"
                          >
                            <Styled(SVGInline)
                              color="#006cff"
                              data-active={true}
                              in="blue"
                              rotate={0}
                              size="s"
                              svg="test-file-stub"
                            >
                              <SVGInline
                                className="c7"
                                classSuffix="-svg"
                                cleanup={Array []}
                                cleanupExceptions={Array []}
                                color="#006cff"
                                component="span"
                                data-active={true}
                                in="blue"
                                rotate={0}
                                size="s"
                                svg="test-file-stub"
                              >
                                <span
                                  className="SVGInline c7"
                                  color="#006cff"
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "test-file-stub",
                                    }
                                  }
                                  data-active={true}
                                  in="blue"
                                  rotate={0}
                                  size="s"
                                />
                              </SVGInline>
                            </Styled(SVGInline)>
                          </Icon>
                        </span>
                      </styled.span>
                    </li>
                  </styled.li>
                  <styled.li
                    key="linkedin"
                    title="linkedin social share"
                  >
                    <li
                      className="c4"
                      title="linkedin social share"
                    >
                      <styled.span
                        onClick={[Function]}
                        tabIndex={1}
                      >
                        <span
                          className="c5"
                          onClick={[Function]}
                          tabIndex={1}
                        >
                          <Icon
                            active={true}
                            color="#006cff"
                            i="test-file-stub"
                            in="blue"
                            rotate={0}
                            size="s"
                          >
                            <Styled(SVGInline)
                              color="#006cff"
                              data-active={true}
                              in="blue"
                              rotate={0}
                              size="s"
                              svg="test-file-stub"
                            >
                              <SVGInline
                                className="c7"
                                classSuffix="-svg"
                                cleanup={Array []}
                                cleanupExceptions={Array []}
                                color="#006cff"
                                component="span"
                                data-active={true}
                                in="blue"
                                rotate={0}
                                size="s"
                                svg="test-file-stub"
                              >
                                <span
                                  className="SVGInline c7"
                                  color="#006cff"
                                  dangerouslySetInnerHTML={
                                    Object {
                                      "__html": "test-file-stub",
                                    }
                                  }
                                  data-active={true}
                                  in="blue"
                                  rotate={0}
                                  size="s"
                                />
                              </SVGInline>
                            </Styled(SVGInline)>
                          </Icon>
                        </span>
                      </styled.span>
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
            </Social>
          </div>
        </styled.div>
      </TopBar>
      <Heading
        scroll={true}
      >
        <styled.div
          scroll={true}
        >
          <div
            className="c8"
          >
            <h1>
              Adele
            </h1>
            <span>
              The repository of publicly available design systems and pattern libraries
            </span>
            <styled.section>
              <section
                className="c9"
              >
                <Button
                  action={[Function]}
                  label="Start Exploring Systems"
                  tab={1}
                  targetBlank={false}
                  type="button"
                >
                  <styled.button
                    onClick={[Function]}
                    tabIndex={1}
                  >
                    <button
                      className="c10"
                      onClick={[Function]}
                      tabIndex={1}
                    >
                      Start Exploring Systems
                    </button>
                  </styled.button>
                </Button>
                <NavLink
                  action={[Function]}
                  label="Why Adele?"
                  tab={1}
                >
                  <styled.button
                    onClick={[Function]}
                    tabIndex={1}
                  >
                    <button
                      className="c11"
                      onClick={[Function]}
                      tabIndex={1}
                    >
                      Why Adele?
                    </button>
                  </styled.button>
                </NavLink>
              </section>
            </styled.section>
          </div>
        </styled.div>
      </Heading>
    </header>
  </styled.header>
</HeaderContainer>
`;
